app/redux-modules/reports.js

```diff
@@ -172,3 +247,22 @@ export function uploadReport({ reportName, fields }) {
     });
   };
 }
+
+export function syncReports() {
+  return (dispatch, state) => {
+    const { reports } = state();
+    if (!reports.synced && !reports.syncing) dispatch(getDefaultReport());
+    const { pendingData } = state().reports;
+    if (getActionsTodoCount(pendingData) > 0) {
+      Object.keys(pendingData).forEach((template) => {
```

Comment:
I would call template differently, because it's not actually a template, it's a dictionary that contains the templates.